File: docs/code-examples/all/scalar_simple.cpp
// Log a scalar over time.

#include <rerun.hpp>

#include <cmath>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_scalar");
    rec.spawn().exit_on_failure();

    // Log the data on a timeline called "step".
    for (int step = 0; step < 64; ++step) {
        rec.set_time_sequence("step", step);
        rec.log("scalar", rerun::Scalar(std::sin(static_cast<double>(step) / 10.0)));
    }
}

File: docs/snippets/all/scalar_simple.py
"""Log a scalar over time."""
import math

import rerun as rr

rr.init("rerun_example_scalar", spawn=True)

# Log the data on a timeline called "step".
for step in range(0, 64):
    rr.set_time_sequence("step", step)
    rr.log("scalar", rr.Scalar(math.sin(step / 10.0)))
